### ISSUE TITLE
Snake conversion is not being done on sql INSERT... for bean fields in camelCase

### ISSUE THREAD

**Issue by gcristian** [jdbi/jdbi#1618]
I dont know when JDBI does the conversion from camel-case bean's mapped fields to database's snake-case fields, or if JDBI does support this at all. 
Using PostgreSQL 10, Lombok, JDBI 3 

This's the DB's DDL:
```
dbx=# \d proxies
                                          Table "public.proxies"
     Column     |            Type             | Collation | Nullable |               Default               
----------------+-----------------------------+-----------+----------+-------------------------------------
 id             | integer                     |           | not null | nextval('proxies_id_seq'::regclass)
 created        | timestamp without time zone |           |          | 
 modified       | timestamp without time zone |           |          | now()
 state_modified | timestamp without time zone |           |          | 
 state          | character varying(20)       |           |          | 
```

The Bean that I'm trying to insert:
```
@JsonInclude(value = Include.NON_NULL)
@Builder(toBuilder = true)
@AllArgsConstructor 
@Getter 
public class Proxy {

	private Instant stateModified ;

	// many other fields
```

The Stacktrace shows the rendered SQL without snakeCase conversion:
You can see `stateModified` not converted to `state_modified`
```
org.jdbi.v3.core.statement.UnableToExecuteStatementException: java.sql.BatchUpdateException: Batch entry 0 INSERT INTO proxies (created, modified, stateModified, state) VALUES ('2019-10-26 20:31:24.392-03', '2019-10-26 20:31:24.392-03', '2019-10-26 20:31:24.392-03', 'ACTIVE')
RETURNING * was aborted: ERROR: column "stateModified" of relation "proxies" does not exist
  Position: 41  Call getNextException to see other errors in the batch. [statement:"INSERT INTO proxies (created, <fields>) VALUES (:created, <values>)", rewritten:"INSERT INTO proxies (created
, modified, stateModified, state) VALUES (:created, :modified, :stateModified, :state)", parsed:"ParsedSql{sql='INSERT INTO proxies (created, modified, stateModified, state) VALUES (?, ?, ?, ?)', parameters=ParsedParameters{positional=false, parameterNames=[created, modified, stateModified, state]}}", arguments:{positional:{}, named:{}, finder:[{lazy bean property arguments ..."]}]
  at org.jdbi.v3.core.statement.PreparedBatch.internalBatchExecute(PreparedBatch.java:179)
  at org.jdbi.v3.core.statement.PreparedBatch.lambda$execute$0(PreparedBatch.java:111)
  at org.jdbi.v3.core.result.ResultProducers.lambda$getGeneratedKeys$4(ResultProducers.java:108)
  at org.jdbi.v3.core.result.ResultIterable.lambda$of$0(ResultIterable.java:53)
  at org.jdbi.v3.sqlobject.statement.internal.SqlBatchHandler.lambda$new$1(SqlBatchHandler.java:84)

```

When debugging: JDBI
Here `renderedSql` has the SQL ready to execute without camelCase conversion to snakeCase:
```
    private ExecutedBatch internalBatchExecute() {
        if (!getBinding().isEmpty()) {
            add();
        }
        if (bindings.isEmpty()) {
            throw new IllegalStateException("No batch parts to execute");
        }

        String renderedSql = getConfig(SqlStatements.class)
                .getTemplateEngine()
                .render(getSql(), getContext());
        getContext().setRenderedSql(renderedSql);

        ParsedSql parsedSql = getConfig(SqlStatements.class)
                .getSqlParser()
                .parse(renderedSql, getContext());
        String sql = parsedSql.getSql();
        ParsedParameters parsedParameters = parsedSql.getParameters();
        getContext().setParsedSql(parsedSql);

```
May be I'm lacking of some custom configuration ? 
This's all the code related to the Repository class and the Bean, at the Guice initialization:

```
		bind(ProxyRepository.class).toInstance(dbi.onDemand(ProxyRepository.class));
```
and:
```
		dbi.installPlugins();
		dbi.registerRowMapper(ConstructorMapper.factory(Proxy.class));
```
This's the ProxyRepository interface method
```
	static String restFields() {
		return "modified, stateModified, state";

	static final String INSERT = "INSERT INTO proxies (created, <fields>) VALUES (:created, <values>)";
	
        @SqlBatch(INSERT)
	@GetGeneratedKeys
	@UseStringTemplateEngine
	@BatchChunkSize(100)
	int[] insert(
			@Define("fields") String fields,
			@Define("values") String values,
			@BindBean Set<Proxy> p);
```	

Basically I'm calling insert(..) with fields and values accordingly with restFields()'s values and place holders for :fieldName.. etc...

**Comment by qualidafial** [comment]
Use camel case when binding bean properties.

Jdbi only converts camel case to snake case when reading data out of the database.

**Comment by qualidafial** [comment]
I'm your example, restFields() should return the snake case column names, and values should use camel case property bindings: `:modified, :stateModified, :state`

**Comment by qualidafial** [comment]
By default, Jdbi will only replace template expressions like `<foo>`, and parameter binding expressions like `:bar`.

See:
* http://jdbi.org/#_templateengine
* http://jdbi.org/#_sqlparser

**Comment by gcristian** [comment]
Thanks for the clarification ! Is easy to expect from the documentation saying "out of the box" support on snake_case <-> camelCase conversion, and I'm clearly out of place.
Why only reading and not writing ? looks pretty easy to implement it on JDBI, but I Must be oversimplifying for sure.

**Comment by qualidafial** [comment]
This misunderstanding is semi common, and we could probably improve the docs to make this clearer.


Anything like `<foo>` or `:foo` in SQL statement is talking to the Jdbi (i.e. Java) side of things, so format those names in camel case to follow Java idioms.

Everything else in the SQL statement is talking to the database, and should be formatted according to database idioms--which in your case means snake case.
